apps/web/lib/video/[uid]/getServerSideProps.ts

```diff
@@ -78,10 +79,21 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
 
   const oldVideoReference = getCalVideoReference(bookingObj.references);
 
+  const endTime = new Date(booking.endTime);
+  const fourteenDaysAfter = new Date(endTime.getTime() + 14 * 24 * 60 * 60 * 1000);
+  const epochTimeFourteenDaysAfter = Math.floor(fourteenDaysAfter.getTime() / 1000);
+
+  const videoReferencePassword = await updateMeetingTokenIfExpired({
+    bookingReferenceId: oldVideoReference.id,
+    roomName: oldVideoReference.uid,
+    meetingToken: oldVideoReference.meetingPassword,
+    exp: epochTimeFourteenDaysAfter,
+  });
+
```

Comment:
If it has reached this part of code then current time must be less than booking.endTime + 14 days (buffer time)  because of this check 

  const isPast = booking?.endTime <= exitDate;
  if (isPast) {
    return {
      redirect: {
        destination: `/video/meeting-ended/${booking?.uid}`,
        permanent: false,
      },
    };
  }